File: FunnyAlgorithms-master/Caesar Cipher/caesar.py

# Caesar cipher implemented over ASCII character set
MAX_KEY_SIZE = 256

def getMode():

    while True:

        print('Do you wish to encrypt or decrypt a message?')

        mode = input().lower()

        if mode in 'encrypt e decrypt d'.split():

            return mode

        else:

            print('Enter either "encrypt" or "e" or "decrypt" or "d".')

def getMessage():

    print('Enter your message:')

    return input()

def getKey():

    key = 0

    while True:

        print('Enter the key number (1-%s)' % (MAX_KEY_SIZE))

        key = int(input())

        if (key >= 1 and key <= MAX_KEY_SIZE):

            return key

def getTranslatedMessage(mode, message, key):

    if mode[0] == 'd':
        return Decrypt(message,key)
    elif mode[0] == 'e':
        return Encrypt(message, key)
    else:
        return "Wrong input"

mode = getMode()
message = getMessage()
key = getKey()

print('Your translated text is:')
print(getTranslatedMessage(mode, message, key))

def Encrypt(message, key):
    cipher_text = ''
    for i in message:
        cipher_text += str((ord(i)+key)%256)
    return cipher_text

def Decrypt(cipher_text, key):
    message = ''
    for i in cipher_text:
        message += str((ord(i)-key)%256)
    return message
        

File: Caesar Cipher/caesar.c
#include <stdio.h>
#include <string.h>

#define SIZE 100

void caesar(char str[], int key){
    for(int i=0; i<strlen(str); i++){
        if(str[i] >= 'a' && str[i] <= 'z'){
            str[i] -= 'a';
            if(str[i]+key<0){
                str[i] += 26;
            }
            str[i] = (str[i]+key)%26+'a';
        }else if(str[i] >= 'A' && str[i] <= 'Z'){
            str[i] -= 'A';
            if(str[i]+key<0){
                str[i] += 26;
            }
            str[i] = (str[i]+key)%26+'A';
        }
    }
    printf("Encrypted text: %s\n", str);
}

int main(){
    char str[SIZE];
    int key, mode;

    printf("Do you wish to 1.decrypt or 2.encrypt a string? (1 or 2): ");
    scanf("%d", &mode);
    fflush(stdin);
    printf("Enter your message: ");
    fgets(str, SIZE, stdin);
    str[strlen(str)-1] = '\0';
    printf("Enter your key: ");
    scanf("%d", &key);

    if(mode == 1){
        key = -key;
    }

    caesar(str, key);
    return 0;
}